src/Sql/Sql/Database Backup/Cmdlet/RestoreAzureRMSqlDatabase.cs

```diff
@@ -398,7 +399,40 @@ protected override AzureSqlDatabaseModel GetEntity()
                 model.Edition = Edition;
             }
 
-            return ModelAdapter.RestoreDatabase(this.ResourceGroupName, restorePointInTime, ResourceId, model);
+            /// get auth headers for cross-sub and cross-tenant restore operations
+            string targetSubscriptionId = ModelAdapter.Context?.Subscription.Id;
+            string sourceSubscriptionId = ParseSourceSubscriptionIdFromResourceId(ResourceId);
+            Dictionary<string, List<string>> auxAuthHeader = null;
+            if (!string.IsNullOrEmpty(ResourceId) && targetSubscriptionId!=sourceSubscriptionId)
+            {
+                List<string> resourceIds = new List<string>();
+                resourceIds.Add(ResourceId);
+                var auxHeaderDictionary = GetAuxilaryAuthHeaderFromResourceIds(resourceIds);
+                if (auxHeaderDictionary != null && auxHeaderDictionary.Count > 0)
+                {
+                    auxAuthHeader = new Dictionary<string, List<string>>(auxHeaderDictionary);
+                }
+            }
+
+            return ModelAdapter.RestoreDatabase(this.ResourceGroupName, restorePointInTime, ResourceId, model, sourceSubscriptionId, auxAuthHeader);
+        }
+
+        /// <summary>
+        /// Parse source subscription id from ResourceId
+        /// </summary>
+        /// <returns>Source Subscription Id</returns>
+        private string ParseSourceSubscriptionIdFromResourceId(string resourceId)
+        {
+            string[] words = resourceId.Split('/');
```

Comment:
check resourceId for null:
if (string.IsNullOrEmpty(resourceId))
{
    return null;
}